### ISSUE TITLE
fetch.lua:67: attempt to index a nil value (local 'fd')

### ISSUE THREAD

**Issue by Zash** [luarocks/luarocks#1667]
### Please fill out the following:
- **Platform:**
  - Linux.

- **LuaRocks version:**
  - 3.11.0

- **Configuration file:**
  - Please try to produce with the default configuration.
  - If you cannot reproduce with the default configuration, please try to find
    the minimal configuration to reproduce.
  - Upload the config to a pastebin such as gist.github.com.

- **LuaRocks output from when the issue occurred:**
  - Use the `--verbose` flag.

- **Description:**
  - The steps you took in plain English to reproduce the problem.

A Prosody user reported this, then left before anyone asked for further details.

> Recently I noticed this error when attempting to check for outdated plugins within my Prosody instance. I have attached a pastebin for the error output:

```
    Error: LuaRocks 3.11.0 bug (please report at https://github.com/luarocks/luarocks/issues).
    Arch.: linux-x86_64
    /usr/share/lua/5.4/luarocks/fetch.lua:67: attempt to index a nil value (local 'fd')
    stack traceback:
    	/usr/share/lua/5.4/luarocks/fetch.lua:67: in function 'luarocks.fetch.fetch_caching'
    	/usr/share/lua/5.4/luarocks/manif.lua:108: in function 'luarocks.manif.load_manifest'
    	/usr/share/lua/5.4/luarocks/search.lua:111: in function </usr/share/lua/5.4/luarocks/search.lua:101>
    	(...tail calls...)
    	/usr/share/lua/5.4/luarocks/search.lua:155: in function 'luarocks.search.search_repos'
    	/usr/share/lua/5.4/luarocks/cmd/list.lua:38: in upvalue 'check_outdated'
    	/usr/share/lua/5.4/luarocks/cmd/list.lua:56: in function </usr/share/lua/5.4/luarocks/cmd/list.lua:54>
    	(...tail calls...)
    	[C]: in function 'xpcall'
    	/usr/share/lua/5.4/luarocks/cmd.lua:765: in function 'luarocks.cmd.run_command'
    	/usr/bin/luarocks:38: in main chunk
    	[C]: in ?

```

> Additional information of note:
> 
> - Prosody 0.12.4
> - Lua 5.4
> - LuaRocks 3.11.0-1
> - LuaFileSystem 1.8.0
> - LuaSec 1.3.2
> - LuaSocket 3.0.0

"attempting to check for outdated plugins" here likely means they ran `prosodyctl list --outdated` which internally executes the following luarocks command:

```shell
luarocks --lua-version=5.4 list --tree=/var/lib/prosody/custom_modules --server=https://modules.prosody.im/rocks/ --outdated
```

Looks like https://github.com/luarocks/luarocks/commit/d84129429a7553211fffd956fc2c05675aa03a06 is missing error handling when opening `checkfile`?

https://github.com/luarocks/luarocks/blob/d84129429a7553211fffd956fc2c05675aa03a06/src/luarocks/fetch.lua#L66-L68

**Comment by hishamhm** [comment]
The fix is merged! Thank you @Zash for the report and @git-hulk for the fix!
